metabase/metabase
# Add embedding options for questions and dashboards

Epic https://github.com/metabase/metabase/issues/22529

Changes:
- Adds `header`,  `additional_info`, `action_buttons` query string params to control full-app embedding behavior
- Moves checks from `App.tsx` to selectors
- Adds e2e tests for changes in this epic

Requirements:
- Set `IFRAMED` to `true` here https://github.com/metabase/metabase/blob/43997d63575e72743f156b3a61ca11aabf4e856e/frontend/src/metabase/lib/dom.js#L13

How to test `header=false`:
- Open `http://localhost:3000?header=false`. 
- Open a saved question. Make sure there is no header.
- Do the same for a draft question, a model, and a dashboard.

<img width="1728" alt="Screenshot 2022-05-11 at 14 08 14" src="https://user-images.githubusercontent.com/8542534/167835732-4c47095c-456a-46b7-8e51-0ddb988e0bbf.png">

How to test `additional_info=false`:
- Open `http://localhost:3000?additional_info=false`. 
- Open a saved question. Make sure there are last edit label and breadcrumbs. 
- Do the same for a model and a dashboard.

<img width="1728" alt="Screenshot 2022-05-11 at 14 08 46" src="https://user-images.githubusercontent.com/8542534/167836072-4f34407a-e54d-40c1-b4dd-9fb591072f2f.png">

How to test `action_buttons=false`:
- Open `http://localhost:3000?action_buttons=false`. 
- Open a saved question. Make sure there are no filter, summarize, notebook buttons.

<img width="1728" alt="Screenshot 2022-05-11 at 14 09 10" src="https://user-images.githubusercontent.com/8542534/167835985-1fdf2957-4976-404f-a37a-9fcb254bada7.png">
